metpy/calc/thermo.py

```diff
@@ -439,17 +443,30 @@ def lfc(pressure, temperature, dewpt, parcel_temperature_profile=None, dewpt_sta
         if not any(idx):
             x, y = this_lcl
             return x, y
-        # Otherwise, make select first candidate LFC above the LCL
+        # Otherwise, find all LFCs that exist above the LCL
+        # What is returned depends on which flag as described in the docstring
         else:
-            x = x[idx]
-            y = y[idx]
-            return x[0], y[0]
+            x, y = _multiple_el_lfc_options(x, y, idx, which)
+            return x, y
+
+
+def _multiple_el_lfc_options(intersect_pressures, intersect_temperatures, valid_x,
+                             which):
+    """Choose which ELs and LFCs to return from a sounding."""
+    p_list, t_list = intersect_pressures[valid_x], intersect_temperatures[valid_x]
+    if which == 'all':
+        x, y = p_list, t_list
+    if which == 'bottom':
```

Comment:
This `if` and the following one could instead be `elif` which stands for "else if". What it means is that once one check passes, it won't run the others.